src/kontrol/prove.py

```diff
@@ -71,6 +74,22 @@ def foundry_prove(
         else []
     )
 
+    if prove_options.cse:
+        test_suite = collect_tests(foundry, tests, reinit=prove_options.reinit, return_empty=True)
+        for test in test_suite:
+            if not isinstance(test.method, Contract.Method) or test.method.function_calls is None:
+                continue
+
+            test_version_tuples = [
+                parse_test_version_tuple(t) for t in test.method.function_calls if t not in summary_ids
+            ]
+
+            if len(test_version_tuples) > 0:
+                print(f'For test {test.name}, found external calls: {test_version_tuples}')
+                summary_ids.extend(
+                    p.id for p in foundry_prove(foundry, prove_options, rpc_options, test_version_tuples)
+                )
```

Comment:
A major issue here is that we'll be proving all subproofs is sequence, instead of taking advantage of our parallel prover abilities. So this will be much slower than in could be because we're not dispatching subproofs in parallel.

Reply:
Here we could have something like
```py
        prove_option.workers = min(len(test_version_tuples), MAX_WORKERS)
```
where `MAX_WORKERS` can be defined to somewhere between 8 or 12.

Reply:
Ah, thanks for the explanation below, I think I understand better now.